client/components/forms/form-text-input/index.jsx

```diff
@@ -1,45 +1,48 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
+import { omit } from 'lodash';
 
-export default React.createClass( {
+export default class FormTextInput extends PureComponent {
+	static propTypes = {
+		isError: PropTypes.bool,
+		isValid: PropTypes.bool,
+		selectOnFocus: PropTypes.bool,
+		className: PropTypes.string
+	};
 
-	displayName: 'FormTextInput',
+	constructor() {
+		super( ...arguments );
 
-	getDefaultProps() {
-		return {
-			isError: false,
-			isValid: false,
-			selectOnFocus: false,
-			type: 'text'
-		};
-	},
+		this.selectOnFocus = this.selectOnFocus.bind( this );
+	}
 
 	focus() {
 		this.refs.textField.focus();
-	},
+	}
+
+	selectOnFocus( event ) {
+		if ( this.props.selectOnFocus ) {
+			event.target.select();
+		}
+	}
 
 	render() {
-		const { className, selectOnFocus } = this.props;
-		const classes = classNames( className, {
-			'form-text-input': true,
+		const props = omit( this.props, 'isError', 'isValid', 'selectOnFocus' );
```

Comment:
>How about:

I've used this pattern in the past, and it's indeed convenient. There's an argument to be made that omitting programmatically allows us to be less intentional with prop passing. Additionally, I'd love for us to use a plugin like [`babel-plugin-transform-react-remove-prop-types`](https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types) to help reduce bundle sizes, but when I'd attempted to use it in the past I found that our references to `propTypes` on the component class directly cause errors. 

Minor correction: you'd need to call `Object.keys` on `FormTextInput.propTypes` before passing to `omit`.